packages/eslint-plugin-query/src/utils/detect-react-query-imports.ts

```diff
@@ -0,0 +1,72 @@
+import type { ESLintUtils, TSESLint, TSESTree } from '@typescript-eslint/utils'
+
+type Create = Parameters<
+  ReturnType<typeof ESLintUtils.RuleCreator>
+>[0]['create']
+
+type Context = Parameters<Create>[0]
+type Options = Parameters<Create>[1]
+type Helpers = {
+  isReactQueryImport: (node: TSESTree.Identifier) => boolean
+}
+
+export type EnhancedCreate = (
+  context: Context,
+  options: Options,
+  helpers: Helpers,
+) => ReturnType<Create>
+
+export function detectReactQueryImports(create: EnhancedCreate): Create {
+  return (context, optionsWithDefault) => {
+    const reactQueryImportSpecifiers: TSESTree.ImportClause[] = []
+
+    const helpers: Helpers = {
+      isReactQueryImport(node) {
+        return !!reactQueryImportSpecifiers.find((specifier) => {
+          if (specifier.type === 'ImportSpecifier') {
+            return node.name === specifier.local.name
+          }
+        })
+      },
+    }
+
+    const detectionInstructions: TSESLint.RuleListener = {
+      ImportDeclaration(node) {
+        if (
+          node.specifiers.length > 0 &&
+          node.importKind === 'value' &&
+          node.source.value === '@tanstack/react-query'
```

Comment:
also here, this wouldn't work for other adapters 🤔

Reply:
I agree. Although, there are other libraries who have `useQuery` as well, such as [Chakra UI's useQuery](https://github.com/chakra-ui/chakra-ui/blob/f0e3fb61e3064f03e64222650cf0f1570843a146/packages/components/media-query/src/media-query.tsx#L13) (I always accidentally import that hook 😅).

I copied the same logic that was in `react-query-eslint-plugin`. I can remove that part.

Reply:
I think checking the import is fine, but it would need to be widened to include all tanstack adapters

Reply:
done